pop_tools/budget.py

```diff
@@ -0,0 +1,129 @@
+import datetime
+import glob
+
+import xarray as xr
+
+from .grid import get_grid
+
+MOLS_TO_MOLYR = 1e-9 * 86400.0 * 365.0
+
+
+def _process_coords(ds, concat_dim='time', drop=True, extra_coord_vars=['time_bound']):
+    """Preprocessor function to drop all non-dim coords, which slows down concatenation.
+
+    Borrowed from @sridge in an issue thread.
+    """
+
+    time_new = ds.time - datetime.timedelta(seconds=1)
```

Comment:
esmlab has a superior way of correcting the time coordinate that I just started using thanks to @matt-long. I would suggest this instead:
```Python
ds = ds.esmlab._ds.esmlab.compute_time_var()
```

Reply:
The way I access that function doesn't seem correct, however. I feel like ```ds.esmlab.compute_time_var()``` should work, but it doesn't

Reply:
> I remember changing dask config to get this working when I was doing budgets with my CESM tracer budget notebook. Try turning off "spill to disk" in your dask config and give these settings a go

@sridge, I've had all these settings set for these tests in `~/.config/dask/jobqueue.yml`. So that doesn't seem to be the problem here.

> You may need to monitor dask workers' memory via the dashboard. It's most likely that the error is also coming from bigger chunks that can't fit in the individual dask worker memory. What's the memory size for an individual worker in the cluster?

I tried the `PBSCluster` command that Sean used in his notebook:
```python
cluster = PBSCluster(
                     processes=18,
                     memory="6GB",
                     queue='premium',
                     resource_spec='select=1:ncpus=36:mem=109G',
                     walltime='1:00:00')
cluster.start_workers(16)
```

So this implies 6GB per worker which should be plenty for the chunk I'm giving it. I can double-check the dashboard and see their memory limits.